### ISSUE TITLE
Virtual Network creation: Provider produced inconsistent result after apply

### ISSUE THREAD

**Issue by chrismarget-j** [Juniper/terraform-provider-apstra#170]
## Error text:
```
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to apstra_datacenter_virtual_network.x[12], provider "provider[\"registry.terraform.io/juniper/apstra\"]" produced an unexpected new
│ value: .ipv4_virtual_gateway: was cty.StringVal("10.200.13.254"), but now null.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to apstra_datacenter_virtual_network.x[12], provider "provider[\"registry.terraform.io/juniper/apstra\"]" produced an unexpected new
│ value: .ipv4_subnet: was cty.StringVal("10.200.13.0/24"), but now null.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
```

Wireshark trace indicates that indeed, the new Virtual Network's subnet and gateway attributes are empty.

The whole create / task poll / read transaction follows:

## Client sends request to create virtual network `10.200.13.0/24` with `async=full`

```
POST /api/blueprints/1c0c455f-6029-4696-9cda-5e7fde1d7ff6/virtual-networks?async=full HTTP/1.1
Host: 18.236.72.182:25859
User-Agent: Go-http-client/1.1
Content-Length: 438
Accept: application/json
Authtoken: <snip>
Content-Type: application/json
Accept-Encoding: gzip

{"dhcp_service":"dhcpServiceDisabled","ipv4_enabled":true,"ipv4_subnet":"10.200.13.0/24","ipv6_enabled":false,"label":"test1013","rt_policy":null,"security_zone_id":"6k8Wmo0n1h5b_Mbnmbc","svi_ips":[],"virtual_gateway_ipv4":"10.200.13.254","virtual_gateway_ipv4_enabled":true,"virtual_gateway_ipv6_enabled":false,"bound_to":[{"access_switch_node_ids":[],"system_id":"IoxdL2VhZ2UoYs7Mub4","vlan_id":1013}],"vn_id":"11013","vn_type":"vxlan"}
```

## Server responds with task ID `3d4ced7b-d375-4d18-972c-4e1d584ac051`

```
HTTP/1.1 202 ACCEPTED
Server: nginx
Date: Thu, 29 Jun 2023 18:25:06 GMT
Content-Type: application/json
Transfer-Encoding: chunked
Connection: keep-alive
Cache-Control: no-cache
Strict-Transport-Security: max-age=63072000; includeSubdomains;
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Content-Security-Policy: default-src 'self'; img-src 'self' data:; font-src 'self' data:; script-src 'self' 'unsafe-inline' 'unsafe-eval'; worker-src 'self' blob:; style-src 'self' 'unsafe-inline'
Referrer-Policy: strict-origin-when-cross-origin

{"task_id": "3d4ced7b-d375-4d18-972c-4e1d584ac051"}
```

## Client pulls task status after completion

```
GET /api/blueprints/1c0c455f-6029-4696-9cda-5e7fde1d7ff6/tasks/3d4ced7b-d375-4d18-972c-4e1d584ac051?async=full HTTP/1.1
Host: 18.236.72.182:25859
User-Agent: Go-http-client/1.1
Accept: application/json
Authtoken: <snip>
Accept-Encoding: gzip
```

## Server response indicates successful task completion, new object ID is `yaZGRj1ZIWz8sQEVfuM`

```
HTTP/1.1 200 OK
Server: nginx
Date: Thu, 29 Jun 2023 18:25:24 GMT
Content-Type: application/json
Transfer-Encoding: chunked
Connection: keep-alive
Cache-Control: no-cache
Strict-Transport-Security: max-age=63072000; includeSubdomains;
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Content-Security-Policy: default-src 'self'; img-src 'self' data:; font-src 'self' data:; script-src 'self' 'unsafe-inline' 'unsafe-eval'; worker-src 'self' blob:; style-src 'self' 'unsafe-inline'
Referrer-Policy: strict-origin-when-cross-origin
Content-Encoding: gzip

{"status": "succeeded", "begin_at": "2023-06-29T18:25:22.466226+0000", "request_data": {"url": "http://18.236.72.182:25859/api/blueprints/1c0c455f-6029-4696-9cda-5e7fde1d7ff6/virtual-networks?async=full", "headers": {"Content-Length": "438", "Accept-Encoding": "gzip", "X-Forwarded-Proto": "https", "X-Forwarded-For": "47.14.45.97", "Connection": "close", "X-User": "admin", "Accept": "application/json", "User-Agent": "Go-http-client/1.1", "Host": "18.236.72.182:25859", "X-User-Id": "6f293397-f82c-47fb-a1a3-0674d039baf3", "X-Real-Ip": "47.14.45.97", "Content-Type": "application/json"}, "args": {"async": "full"}, "data": {"virtual_gateway_ipv4_enabled": true, "bound_to": [{"access_switch_node_ids": [], "system_id": "IoxdL2VhZ2UoYs7Mub4", "vlan_id": 1013}], "dhcp_service": "dhcpServiceDisabled", "virtual_gateway_ipv4": "10.200.13.254", "vn_id": "11013", "label": "test1013", "virtual_gateway_ipv6_enabled": false, "vn_type": "vxlan", "rt_policy": null, "ipv6_enabled": false, "ipv4_subnet": "10.200.13.0/24", "svi_ips": [], "ipv4_enabled": true, "security_zone_id": "6k8Wmo0n1h5b_Mbnmbc"}, "method": "POST"}, "user_id": "6f293397-f82c-47fb-a1a3-0674d039baf3", "last_updated_at": "2023-06-29T18:25:23.917954+0000", "user_name": "admin", "created_at": "2023-06-29T18:25:06.344884+0000", "detailed_status": {"api_response": {"id": "yaZGRj1ZIWz8sQEVfuM"}, "config_blueprint_version": 1754}, "config_last_updated_at": "2023-06-29T18:25:06.344884+0000", "user_ip": "47.14.45.97", "type": "blueprint.facade.POST./virtual-networks", "id": "3d4ced7b-d375-4d18-972c-4e1d584ac051"}
```

## Client requests object `yaZGRj1ZIWz8sQEVfuM`

```
GET /api/blueprints/1c0c455f-6029-4696-9cda-5e7fde1d7ff6/virtual-networks/yaZGRj1ZIWz8sQEVfuM HTTP/1.1
Host: 18.236.72.182:25859
User-Agent: Go-http-client/1.1
Accept: application/json
Authtoken: <snip>
Accept-Encoding: gzip
```

## Server responds with object details. The `ipv4_subnet` and `virtual_gateway_ipv4` fields do not contain the expected values, but rather `null`:

```
HTTP/1.1 200 OK
Server: nginx
Date: Thu, 29 Jun 2023 18:25:24 GMT
Content-Type: application/json
Transfer-Encoding: chunked
Connection: keep-alive
Cache-Control: no-cache
Strict-Transport-Security: max-age=63072000; includeSubdomains;
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Content-Security-Policy: default-src 'self'; img-src 'self' data:; font-src 'self' data:; script-src 'self' 'unsafe-inline' 'unsafe-eval'; worker-src 'self' blob:; style-src 'self' 'unsafe-inline'
Referrer-Policy: strict-origin-when-cross-origin
Content-Encoding: gzip

{"bound_to": [{"access_switch_node_ids": [], "system_id": "IoxdL2VhZ2UoYs7Mub4", "vlan_id": 1013}], "svi_ips": [{"ipv6_addr": null, "ipv4_mode": "enabled", "ipv4_requirement": null, "ipv4_addr": null, "ipv6_requirement": null, "system_id": "IoxdL2VhZ2UoYs7Mub4", "ipv6_mode": "disabled"}], "rt_policy": null, "id": "yaZGRj1ZIWz8sQEVfuM", "dhcp_service": "dhcpServiceDisabled", "virtual_gateway_ipv4": null, "virtual_gateway_ipv6": null, "floating_ips": [], "label": "test1013", "virtual_gateway_ipv6_enabled": false, "reserved_vlan_id": null, "ipv4_enabled": true, "description": null, "virtual_mac": null, "ipv6_subnet": null, "vn_type": "vxlan", "ipv4_subnet": null, "ipv6_enabled": false, "security_zone_id": "6k8Wmo0n1h5b_Mbnmbc", "route_target": "11013:1", "virtual_gateway_ipv4_enabled": true, "vn_id": "11013", "endpoints": []}
```

A short time later, a subsequent query finds the empty fields populated with the expected values.

Per Apstra engineering, task completion isn't tied to resource allocation. Any of the following could produce this sort of condition:

- Virtual network - subnets, gateway IPs, SVIs, VXLAN L2 VNI
- Routing zones - L3 VNIs
- Physical links, logical links - subets
- ASNs
- Loopback IPs

It's not immediately clear what we should do about it.

**Comment by chrismarget-j** [comment]
branch bug/170 attempts to mitigate this issue by copying the VNI, IPv4 subnet+gateway and IPv6 subnet+gateway config values over the API response values when those values are set in the config.

**Comment by chrismarget-j** [comment]
Additional context in AOS RFE-2978

**Comment by chrismarget-j** [comment]
Closed by #179
